icicle-source/src/Icicle/Source/Transform/ReifyPossibility.hs

```diff
@@ -296,7 +296,15 @@ makeApps a fun (arg:rest) doWrap
 
  -- If argument is a definitely, just apply it as usual
  | otherwise
- =  makeApps a (App a fun arg) rest doWrap
+ =  makeApps a (App annotWrapPrim fun arg) rest doWrap
```

Comment:
Because there were no primitives that returned Possibly before, the return type of the application itself never needed to change - it was being unwrapped and rewrapped, so the primitives were still operating over real, definite values. Now that `map_insert` returns Possibly, we need to change the type annotation for applications to include the Sum.